node/src/blockchain/blockchain_bridge.rs

```diff
@@ -667,70 +717,125 @@ mod tests {
+            )));
+        let consuming_wallet = make_wallet("somewallet");
+        let persistent_configuration_mock =
+            PersistentConfigurationMock::new().gas_price_result(Ok(3u64));
+        let subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_mock),
+            Box::new(persistent_configuration_mock),
+            false,
+            Some(consuming_wallet),
+        );
+        let request = ReportAccountsPayable {
+            accounts: vec![PayableAccount {
+                wallet: make_wallet("blah"),
+                balance_wei: 123_456,
+                last_paid_timestamp: SystemTime::now(),
+                pending_payable_opt: None,
+            }],
+            response_skeleton_opt: None,
+        };
+
+        let result = subject.process_payments(&request);
+
+        assert_eq!(
+            result,
+            Err(PayablePaymentError::TransactionCount(
+                BlockchainError::QueryFailed("What the hack...??".to_string())
+            ))
+        );
+    }
+
+    #[test]
+    fn process_payments_returns_error_from_sending_batch() {
+        let transaction_hash = make_tx_hash(789);
+        let blockchain_interface_mock = BlockchainInterfaceMock::default()
+            .get_transaction_count_result(Ok(web3::types::U256::from(1)))
+            .send_payables_within_batch_result(Err(PayablePaymentError::Sending {
+                msg: "failure from exhaustion".to_string(),
+                hashes: vec![transaction_hash],
+            }));
+        let consuming_wallet = make_wallet("somewallet");
+        let persistent_configuration_mock =
+            PersistentConfigurationMock::new().gas_price_result(Ok(3u64));
+        let mut subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_mock),
+            Box::new(persistent_configuration_mock),
+            false,
+            Some(consuming_wallet.clone()),
+        );
+        let request = ReportAccountsPayable {
+            accounts: vec![PayableAccount {
+                wallet: make_wallet("blah"),
+                balance_wei: 424_454,
+                last_paid_timestamp: SystemTime::now(),
+                pending_payable_opt: None,
+            }],
+            response_skeleton_opt: None,
+        };
+        let (accountant, _, _) = make_recorder();
+        let fingerprint_recipient = accountant.start().recipient();
+        subject
+            .paid_payable_confirmation
+            .new_pp_fingerprints_sub_opt = Some(fingerprint_recipient);
+
+        let result = subject.process_payments(&request);
+
+        assert_eq!(
+            result,
+            Err(PayablePaymentError::Sending {
+                msg: "failure from exhaustion".to_string(),
+                hashes: vec![transaction_hash]
+            })
+        );
     }
 
     #[test]
-    fn handle_report_account_payable_manages_gas_price_error() {
+    fn handle_report_accounts_payable_manages_gas_price_error() {
```

Comment:
I don't know enough about alternative blockchains. I wonder how common the concept of a gas price is. If it's an Ethereum-only thing, we should probably find a way to abstract it out of everything but the BlockchainInterface. However, every blockchain I know of, including Bitcoin, allows or requires payments of some sort to the miners.

Anyway, something to think about from a design point of view. If we decide to switch away from Ethereum--or even keep Ethereum and add another (completely different) blockchain, we'll see how many unintended dependencies we have on Ethereum.